developer/tutorial/tutorial-debugging-server-code.md

```diff
@@ -0,0 +1,145 @@
+
+Since you are currently debugging the Reaction server, you'll only have access to code that runs on the server - not the client.
+
+4. Now open `localhost:3000` as you normally would, and the code should stop executing at your first breakpoint.
+
+In this example, the code stopped executing at a breakpoint in the Products publication, which gets called whenever a client connects to the product grid on the index route:
+
+![DevTools setup](https://blog.reactioncommerce.com/content/images/2017/11/devtools-variables.png)
+
+5. At this breakpoint, you can access the Console by hitting `esc` and opening the Drawer.
+
+![DevTools setup](http://g.recordit.co/1hRutFIaLe.gif)
+
+Now you can watch variables, check the call stack and investigate scope to better debug while you develop.
+
+To learn more about Chrome DevTools, check out Google's [Tools for Web Developers](https://developers.google.com/web/tools/chrome-devtools/javascript/).
+
+## Inspecting in VSCode
+
+_Note:_ You can only have one debugger connected at a time, so if you've already connected to DevTools, then you'll need to disconnect before you can connect with VS Code.
+
+Setting up [VS Code](https://code.visualstudio.com/) and connecting it to the Node debugger is only slightly more complicated than using DevTools. But once it's set up, it can easily become a part of your regular workflow.
+
+1. In the root of your project directory, add a `.vscode/launch.json` file.
+
+![DevTools setup](https://blog.reactioncommerce.com/content/images/2017/11/vscode-launch.png)
+
+2. Set up your file:
+
+*.vscode/launch.json*
+```js
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "Reaction Server",
+            "cwd": "${workspaceRoot}/",
+            "runtimeExecutable": "${workspaceRoot}/.meteor/local/dev_bundle/bin/npm",
+            "restart": true,
+            "timeout": 30000,
+            "stopOnEntry": false,
+            "sourceMaps": true,
+            "protocol": "inspector",
+            "port": 9229
+        }
+    ]
+}
+```
+This borrows heavily from a Meteor forum post on [Meteor 1.6 server debugging with VS Code](https://forums.meteor.com/t/meteor-1-6-server-debugging-with-vs-code/39821).
+
+**Tip:** Port 9229 is the default Node inspector port, but if you switch to another one, eg. `--inspect=5000`, then you'll need to adjust the port in your `launch.json` file.
+
+3. Open the debug panel and click the *Play* icon
+
+![DevTools setup](https://blog.reactioncommerce.com/content/images/2017/11/vscode-devtools.png)
+
+Now you can debug without even leaving your code editor.
+
+To learn more about debugging JavaScript with VS Code, check out [VS Code's debugging guide](https://code.visualstudio.com/docs/nodejs/nodejs-debugging).
+
+## Inspecting in Webstorm
+
+1. Create a new JavaScript Debug configuration.
```

Comment:
1. Create a new Run/Debug configuration based on the "Meteor" default

Then
1. Select your breakpoints by clicking along the left hand side
1. Click on the `debug` icon to start your Reaction app in debug
1. Use the "step in/step out/step over" buttons to navigate through your code.